Assets/AirConsole/examples/pong/ExamplePongLogic.cs

```diff
@@ -1,129 +1,240 @@
+    {
+        Time.timeScale = 0;
+        // If we were playing any sounds we must mute them when an ad is shown
+        AudioListener.pause = true;
+    }
+
+    /// <summary>
+    /// Resume the game after the Ad completed
+    /// </summary>
+    /// <param name="adShown"></param>
+    private void OnAdComplete(bool adShown)
+    {
+        Time.timeScale = 1;
+        AudioListener.pause = false;
+    }
+
+    /// <summary>
+    ///     Updates the displayed text on the screen and sets the device state property "enough_players"
+    ///     which allows the controller to react to the new device state
+    /// </summary>
+    private void CheckTwoPlayers()
+    {
+        var connectedControllers = AirConsole.instance.GetControllerDeviceIds();
+
+        // Only update if the game didn't have active players.
+        if (AirConsole.instance.GetActivePlayerDeviceIds.Count == 0)
+        {
+            if (connectedControllers.Count > 2)
+                uiText.text = "Only 2 players can play, sorry";
+            else if (connectedControllers.Count == 2)
+                uiText.text = "Ready to start";
+            else if (connectedControllers.Count == 1)
+                uiText.text = "Need 1 more player!";
+            else
+                uiText.text = "Need 2 more players!";
+
+            AirConsole.instance.SetCustomDeviceStateProperty("enough_players", connectedControllers.Count == 2);
+        }
+    }
+
+    /// <summary>
+    ///     Sends a message to vibrate for a certain duration to the specific player device.
+    /// </summary>
+    /// <param name="player"></param>
+    private void Vibrate(int player)
+    {
+        AirConsole.instance.Message(AirConsole.instance.ConvertPlayerNumberToDeviceId(player), new { vibrate = 1000 });
+    }
+
+    /// <summary>
+    ///     Sets the device state property "screen" to the given string
+    /// </summary>
+    /// <param name="newScreen"></param>
+    private void SetGameScreen(string newScreen)
+    {
+        screen = newScreen;
+        AirConsole.instance.SetCustomDeviceStateProperty("screen", screen);
+    }
+
+    private void StartGame()
+    {
+        SetGameScreen(GAME);
+        AirConsole.instance.SetActivePlayers(2);
+        ResetBall(true);
+        scoreRacketLeft = 0;
+        scoreRacketRight = 0;
+        UpdateScoreUI();
+    }
+
+    private void BackToLobby()
+    {
+        SetGameScreen(LOBBY);
+        AirConsole.instance.SetActivePlayers(0);
+        ResetBall(false);
+        racketLeft.velocity = Vector2.zero;
+        racketRight.velocity = Vector2.zero;
+        scoreRacketLeft = 0;
+        scoreRacketRight = 0;
+        CheckTwoPlayers();
+    }
+
+    private void ResetBall(bool move)
+    {
+        // place ball at center
+        ball.position = Vector3.zero;
+
+        // push the ball in a random direction
+        if (move)
+        {
+            var startDir = new Vector3(Random.Range(-1, 1f), Random.Range(-0.1f, 0.1f), 0);
```

Comment:
explicit type please - allows you to use the typeless initializer.

```suggestion
            Vector3 startDir = new(Random.Range(-1, 1f), Random.Range(-0.1f, 0.1f), 0);
```